thoth/common/openshift.py

```diff
@@ -1608,6 +1609,43 @@ def get_workflow(
 
         return wf
 
+    def get_workflow_status_report(
+        self,
+        workflow_id: Optional[str] = None,
+        label_selector: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ) -> Tuple[Dict[str, Any], int]:
+        """Get workflow status report, derived from workflow status."""
+        if workflow_id is None:
+            return {"error": "No analysis id provided"}, 400
+        try:
+            wf_status = self.get_workflow_status(
+                name=workflow_id, label_selector=label_selector, namespace=namespace
+            )
+            status = {}
+            # the Job has not started (yet) but the Workflow has been submitted
+            # if the Workflow fails, the status will contain the finished
+            # time of the workflow
+            status["status"], code = (
+                {
+                    "finished_at": wf_status.get("finishedAt"),
+                    "reason": None,
```

Comment:
Why is `reason` included in the status if it will always be set to None.

Reply:
This is to make it compatible to the existing [status report](ttps://github.com/thoth-station/user-api/blob/c1b80baae49b90556534f3c5de3c5e47a0c8d530/thoth/user_api/api_v1.py#L360).

Reply:
Do we want to add some info here (e.g. why a workflow failed) or do we want to completely discard this item?

Reply:
@fridex Maybe if we comb through all the nodes, we can find info like which workflows failed or something?
If the particular step has finished and failed, it has a [message key](https://gist.github.com/saisankargochhayat/6ed94ee54f8221053fd2f1a29f560e63#file-adviser_status-json-L116), we could iterate through nodes to see if such a message exists? I am not sure if the message is same for all. 
This is what I found from the documentation - https://argoproj.github.io/argo/swagger/#ioargoprojworkflowv1alpha1workflowstatus
However all the keys aren't there in the dict we are receiving, like `message` in the root level.

Reply:
This might be tricky to implement for users of Thoth who do not see/don't want to know details about workflows. As workflows can fail in many places (and still be successful based on the configuration like skips on an error), let's keep it as None for now for simplicity (or some generic message). Some of these messages are not useful for end-users, like `failed with exit code 1` or `No more retries left`.

We can check what errors we observe and if we want to populate them to users (with a descriptive error message).

WDYT?

Reply:
Makes sense, let's keep it none for now, however let's have key in-case we want to pass more information in the future.